### PR TITLE
Added AdvancedSlider widget with demo example

### PR DESCRIPTION
**Pull request by MEisebitt**
An advanced version of the slider, that displays the value and allows for keyboard input.

### CODE REVIEW

**Review by derekdreery — APPROVED** [review]
Looks good.

**Comment by derekdreery** [comment]
Do you wanna fix up the clippy warnings? It's not a requirement (I can merge anyway) but it might be neater. :)

File: src/advanced_slider.rs [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+            let rounded_box = RoundedRect::new(2.0, 2.0, 122.0, 22.0, 2.0);
+            ctx.fill(rounded_box, &Color::rgb8(50, 50, 50));
+            ctx.stroke(rounded_box, &Color::rgb8(80, 80, 80), 1.0);
+
+            // Center Text and draw it
+            let text_width = self.val_text.layout_metrics().size.width;
+            self.val_text.draw(
+                ctx,
+                Point::new(62.0 - (text_width / 2.0), 2.0 + self.text_offset),
+            );
```
Comment (totsteps) [inline src/advanced_slider.rs:321]: Looks great.

Curious whether having this outside if-else would make a difference?

**Comment by MEisebitt** [comment]
Yes, I will clean it up :)
How do I run clippy for my example, do I need to do it in some special way ?

File: src/advanced_slider.rs (hunk 1 above)
Comment (MEisebitt) [reply src/advanced_slider.rs:321]: I think it would not, so I'll change that as well.
Thanks for the hint.

**Comment by derekdreery** [comment]
No worries! Just running `cargo clippy` should be enough I think :)

**Comment by MEisebitt** [comment]
Okay, now everything should be neat Clippy wise :)
Also incorporated @totsteps suggestion.

**Comment by derekdreery** [comment]
Awesome thanks.

**Pull request merged**